src/macosMain/kotlin/co/touchlab/xcode/cli/util/CrashHelper.kt

```diff
@@ -0,0 +1,76 @@
+package co.touchlab.xcode.cli.util
+
+import co.touchlab.kermit.LogWriter
+import co.touchlab.kermit.Severity
+import kotlinx.coroutines.CompletableDeferred
+import kotlinx.coroutines.runBlocking
+import platform.Foundation.*
+import platform.Foundation.NSURLSessionConfiguration.Companion.defaultSessionConfiguration
+
+class CrashHelper : LogWriter() {
+    private val logEntries = mutableListOf<LogEntry>()
+
+    override fun log(severity: Severity, message: String, tag: String, throwable: Throwable?) {
+        logEntries.add(LogEntry(severity, message, tag, throwable))
+    }
+
+    fun upload(e: Throwable) {
+        upload(capture(e))
+    }
+
+    private fun upload(crashReport: String) {
+        val url: NSURL = NSURL.URLWithString("https://touchlab.dev/crash/report")!!
+        val request = NSMutableURLRequest.requestWithURL(url).apply {
+            setHTTPMethod("POST")
+            setValue("text/plain", "Content-Type")
+            setHTTPBody((crashReport as NSString).dataUsingEncoding(NSUTF8StringEncoding))
+        }
+
+        val uploadComplete = CompletableDeferred<Unit>()
+        val session = NSURLSession
+            .sessionWithConfiguration(defaultSessionConfiguration)
+            .dataTaskWithRequest(request) { _, _, _ ->
+                uploadComplete.complete(Unit)
+            }
+
+        runBlocking {
+            session.resume()
+            uploadComplete.await()
+        }
+    }
+
+    private fun capture(e: Throwable): String {
+        var out = ""
```

Comment:
Well, I think making string collections and using `joinToString` is `pp`. However, I do think we should avoid using += on strings if it's more than a couple. In this case, `StringBuilder` would be OK as well. However, since they're already in a List, I agree.
```
val out = StringBuilder()

        if (!logEntries.isEmpty()) {
            out.append("BREADCRUMBS\n===========\n\n")
            out.append(
                logEntries.joinToString("\n")
            )
            out.append("\n\n")
        }

        out.append("FINAL CRASH\n===========\n\n")
        out.append(toStacktraceString(e))
```
Something like that.

Reply:
We could also use the builder function `buildString` to wrap it all together, but that's minor.

Reply:
done, thanks!